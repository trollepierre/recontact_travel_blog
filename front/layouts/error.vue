<template>
  <div class="container">
    <h1 v-if="error.statusCode === 404">
      Page not found
    </h1>
    <h1 v-else>
      An error occurred
    </h1>
    <nuxt-link to="/">
      Home page
    </nuxt-link>
  </div>
</template>

<script>
  export default {
    props: { error: { type: Object, default: () => {} } },
    mounted() {
      if (this.$route.params.id && this.$route.params.id.substring(0, 17) === '/apple-touch-icon-') {
        this.$router.go('/apple-touch-icon.png')
      }
    },
  }
</script>
