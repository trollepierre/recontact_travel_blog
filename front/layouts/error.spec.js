import error from './error.vue'

describe('error layout', () => {
  it('should return page not found when 404', () => {
    console.error = jest.fn()

    const wrapper = shallowMount(error, { propsData: { error: { statusCode: 404 } } })

    expect(wrapper).toMatchSnapshot()
  })

  it('should return error occured not found when not 404', () => {
    console.error = jest.fn()

    const wrapper = shallowMount(error, { propsData: { error: { statusCode: 500 } } })

    expect(wrapper).toMatchSnapshot()
  })
})
