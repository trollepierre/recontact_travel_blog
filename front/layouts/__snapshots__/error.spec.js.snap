// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error layout should return error occured not found when not 404 1`] = `
<div class="container">
  <h1>
    An error occurred
  </h1>
  <nuxt-link to="/">
    Home page
  </nuxt-link>
</div>
`;

exports[`error layout should return page not found when 404 1`] = `
<div class="container">
  <h1>
    Page not found
  </h1>
  <nuxt-link to="/">
    Home page
  </nuxt-link>
</div>
`;
